transport/ice/IceSession.cpp

```diff
@@ -1150,8 +1153,16 @@ void IceSession::CandidatePair::send(const uint64_t now)
         state = InProgress;
     }
     _transactions.push_back(transaction);
-    if (_transactions.size() > 50)
+    if (_transactions.size() > 10)
     {
+        if (localEndpoint.endpoint)
+        {
+            auto& frontTransaction = _transactions.front();
+            if (!frontTransaction.acknowledged())
+            {
+                localEndpoint.endpoint->cancelStunTransaction(frontTransaction.id.get());
+            }
+        }
         _transactions.erase(_transactions.begin());
```

Comment:
As far as I understood. We don't erase elements in the middle, we always delete from `begin` if we need. Then `std::deque` seems more suitable for this case as it can handle the erase on beginning without copy all other elements (when `std::vector` has to move the others to begining).

The down size is `std::queue`on libc++ allocates space for 16 elements when needs only one. `std::vector` grows more granularly but when we have 10 transaction, `std::vector` probably already had allocated space to 16 elements anyway

Reply:
Yes, I will change to deque